### PR TITLE
feat: first time screen UI in demo app

### PR DESCRIPTION
**Pull request by khairul-alam-licon**
# Description
- In Demo App: Add a screen which is displayed before downloading/launching a Mini App. The screen should be displayed only the first time that a particular Mini App is launched.
- The screen should display the Mini App's name, icon, and version.
- The screen should have buttons for "Accept" and "Cancel".
- "Accept" should download and launch the mini app.
- "Cancel" should return to listing screen without downloading the mini app.
 
## Links
- MINI-2499

# Checklist
- [x] I have read the [contributing guidelines](../.github/CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](../.github/CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `./gradlew check` without errors

### CODE REVIEW

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt [hunk 1]
```diff
@@ -129,14 +132,9 @@ class MiniAppDisplayActivity : BaseActivity() {
                 AppSettings.instance.urlParameters
```
Comment (khairul-alam-licon) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:132]: Do I also need to launch first-time screen when displaying miniapp by url?

Comment (corycaywood) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:132]: Hmm, I think probably not. The mini app from a URL won't have a manifest, so we won't be able to display the first time permissions.

File: testapp/src/main/res/layout/window_first_time_miniapp.xml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+        android:layout_alignParentStart="true"
+        android:padding="@dimen/small_8"
+        android:text="@string/miniapp_first_time_tutorial"
+        android:textSize="@dimen/text_large_16" />
+
+    </RelativeLayout>
+
+    <ImageView
+      android:id="@+id/firstTimeAppIcon"
+      android:layout_width="@dimen/miniapp_icon_size"
```
Comment (corycaywood) [inline testapp/src/main/res/layout/window_first_time_miniapp.xml:42]: Maybe we could display the icon a little bigger on this screen. It looks a little too small when it's centered by itself. What do you think?

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/firsttime/FirstTimeLaunch.kt [hunk 3]
```diff
@@ -0,0 +1,9 @@
+package com.rakuten.tech.mobile.testapp.ui.display.firsttime
+
+import androidx.annotation.Keep
+
+@Keep
+data class FirstTimeLaunch(
```
Comment (minh-rakuten) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/firsttime/FirstTimeLaunch.kt:6]: I think you can just simply use the SharedPreference with `key: String, value: Boolean`.

**Review by minh-rakuten — COMMENTED** [review]
The storage of first time permission in sample app should be simplified as possible.

File: testapp/src/main/res/layout/window_first_time_miniapp.xml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+      android:text="@string/action_first_time_accept"
+      android:textColor="@android:color/white" />
+
+    <Button
+      android:id="@+id/firstTimeCancel"
+      android:textStyle="bold"
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:layout_marginTop="@dimen/small_8"
+      android:layout_marginStart="@dimen/large_32"
```
Comment (corycaywood) [inline testapp/src/main/res/layout/window_first_time_miniapp.xml:84]: For me, it's a little bit strange looking that the "Cancel" button is slightly smaller than the "Accept" button. Maybe we could do this for the "Cancel" button instead:
- Remove the background color and shadow
- Make the text color red

So basically the button would just look like red text. What do you think?

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt [hunk 5]
```diff
@@ -129,14 +132,9 @@ class MiniAppDisplayActivity : BaseActivity() {
...
-            viewModel.obtainMiniAppDisplay(
-                this@MiniAppDisplayActivity,
-                appInfo,
-                appId!!,
-                miniAppMessageBridge,
-                miniAppNavigator,
-                AppSettings.instance.urlParameters
-            )
+            val firstTimeWindow = FirstTimeWindow(this@MiniAppDisplayActivity, this)
+            if (appInfo != null) firstTimeWindow.initiate(appInfo, "")
```
Comment (corycaywood) [inline testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:136]: Why do you need to check if `appInfo` is null here? Couldn't you just pass both in a single call? `firstTimeWindow.initiate(appInfo, appId!!)`

**Review by corycaywood — APPROVED** [review]
Looks good! 👍  I just left a couple suggestions about the UI design.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt (hunk 5 above)
Comment (khairul-alam-licon) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/MiniAppDisplayActivity.kt:136]: Oh right! 
I could just simple pass in single call. I'll update this part, Thanks.

File: testapp/src/main/res/layout/window_first_time_miniapp.xml (hunk 2 above)
Comment (khairul-alam-licon) [reply testapp/src/main/res/layout/window_first_time_miniapp.xml]: Yeah, I have changed to `@dimen/miniapp_logo_size` = `86dp`.

File: testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/firsttime/FirstTimeLaunch.kt (hunk 3 above)
Comment (khairul-alam-licon) [reply testapp/src/main/java/com/rakuten/tech/mobile/testapp/ui/display/firsttime/FirstTimeLaunch.kt]: Yeah, Thanks for the feedback, I have updated this part.

File: testapp/src/main/res/layout/window_first_time_miniapp.xml (hunk 4 above)
Comment (khairul-alam-licon) [reply testapp/src/main/res/layout/window_first_time_miniapp.xml:86]: Sure, I have updated this part.

**Review by minh-rakuten — APPROVED** [review]
Looks good.

**Comment by khairul-alam-licon** [comment]
Thank you for reviewing. Last change was use "Cancel" text. 
Let me to merge this for now.

**Pull request merged**